src/base1/cockpit.js

```diff
@@ -4328,6 +4328,16 @@ function factory() {
         var self = this;
         event_mixin(self, { });
 
+        var api = cockpit.dbus(null, { bus: "internal" }).proxy("cockpit.Superuser", "/superuser");
+        api.addEventListener("changed", maybe_reload);
+
+        function maybe_reload() {
+            if (api.valid && self.allowed !== null) {
+                if (self.allowed != (api.Current != "none"))
+                    window.location.reload(true);
```

Comment:
Can this look at the page's manifests, so that pages can declare "I watch out for superuser changes and update myself without reloading"?

Reply:
My plan was that cockpit.permission will be deprecated completely (it never really did all that it was promising). The way for a page to declare that it doesn't want to be reloaded is by stopping to use cockpit.permission.